src/GitRepository.cc

```diff
@@ -0,0 +1,415 @@
+{
+    //load index of last commit
+    LibGitPointer<git_index> gindex{repository_index(repo_.get())};
+
+    //remove files from directory
+    int error = git_index_remove_directory(gindex.get(), seq_directory.c_str(), 0);
+    if (error) throw task::Error("Cannot remove sequence files.");
+    // remove sequence directory from git
+    error = git_index_remove_bypath(gindex.get(), (std::string(seq_directory)+"/").c_str());
+    if (error) throw task::Error("Cannot remove sequence directory.");
+
+    git_index_write(gindex.get());
+    //git_index_write_tree(&tree, gindex.get());
+
+    // delete sequence
+    std::filesystem::remove_all(repo_path_ / seq_directory);
+}
+
+
+git_commit* GitRepository::get_commit(int count)
+{
+    std::string ref = "HEAD^" + std::to_string(count);
+    return get_commit(ref);
+}
+git_commit* GitRepository::get_commit(const std::string& ref)
+{
+    git_commit * commit;
+    git_oid oid_parent_commit;
+
+    // resolve HEAD into a SHA1
+    int error = git_reference_name_to_id( &oid_parent_commit, repo_.get(), ref.c_str());
+    if (error) throw task::Error("Cannot find HEAD of branch.");
+
+    // find commit object by commit ID
+    error = git_commit_lookup( &commit, repo_.get(), &oid_parent_commit );
+    if (error) throw task::Error("Cannot find HEAD of branch.");
+
+    return commit;
+}
+git_commit* GitRepository::get_commit()
+{
+    return get_commit(std::string{"HEAD"});
+}
+
+
+std::vector<FileStatus> GitRepository::collect_status(git_status_list *status) const
+{
+    // get number of submodules
+    const size_t nr_entries = git_status_list_entrycount(status);
+
+    // declare status holding vector for each submodule
+    std::vector<FileStatus> return_array;
+    FileStatus filestats;
+
+    for (size_t i = 0; i < nr_entries; ++i)
+    {
+        const git_status_entry *s = nullptr;
+        const char *old_path = nullptr;
+        const char *new_path = nullptr;
+
+        std::string istatus = "";
+        std::string wstatus = "";
+
+        s = git_status_byindex(status, i);
+
+        // list files which exists but are untouched since last commit
+        //############################################################
+        if (s->status == GIT_STATUS_CURRENT)
+        {
+            filestats.handling = "unchanged";
+            filestats.changes = "unchanged";
+
+            old_path = s->head_to_index->old_file.path;
+            new_path = s->head_to_index->new_file.path;
+
+            filestats.path_name = old_path ? std::string{old_path} : std::string{new_path};
+
+            return_array.push_back(filestats);
+
+            continue;
+        }
+
+        if (s->status & GIT_STATUS_WT_MODIFIED)
+            wstatus = "modified";
+        if (s->status & GIT_STATUS_WT_DELETED)
+            wstatus = "deleted";
+        if (s->status & GIT_STATUS_WT_RENAMED)
+            wstatus = "renamed";
+        if (s->status & GIT_STATUS_WT_TYPECHANGE)
+            wstatus = "typechange";
```

Comment:
Maybe this if-litany could be moved into a function to clean this up a little? Same below.